guide/creating_element/testing.md

```diff
@@ -159,13 +144,64 @@ end
+
+```elixir
+defmodule Example.Pipeline do
+  use Membrane.Pipeline
+
+  def handle_init(options) do
+    state = %{response: options.response}
+    spec = %Membrane.Pipeline.Spec{children: [], links: []}
+    {{:ok, spec}, state}
+  end
+
+  def handle_other({:request, from}, state) do
+    send(from, {:response, state.response})
+    {:ok, state}
+  end
+
+  def handle_other(_message, state),
+    do: {:ok, state}
+end
+``````
+Note that `Membrane.Pipeline.Spec` of our pipeline module can contain empty lists of children and links.
+In order to use callbacks defined in `Example.Pipeline` and pass custom initialization arguments, we have to specify it in `Options`.
+
+```elixir
+Pipeline.start_link(%Pipeline.Options{
+  ...
+  module: Example.Pipeline,
+  custom_args: %{response: "Hello there!"}
+})
+```
+
+Now we can create another test case to try out new functionality.
+
+```elixir
+describe "Decoding Pipeline should" do
+  ...
+
+  test 'Answer with {:reponse, "Hello there!"} message' do
+    assert {:ok, pid} =  Pipeline.start_link(%Pipeline.Options{...})
+    assert Pipeline.play(pid) == :ok # Start the pipeline
+    test_process_pid = self()
+    send(pid, {:request, test_process_pid})
+    assert_pipeline_receive(pid, {:request, test_process_pid}) # Check if pipeline got message
+    assert_receive({:response, "Hello there!"}) # Check if we got return message
+  end
+end
+```
+
 ## Using other testing utilities
 
-Apart from the [`Membrane.Testing.Pipeline`](https://hexdocs.pm/membrane_core/Membrane.Testing.Pipeline.html), which we've already seen, there are a bunch of other testing utilities which may come in handly for different test scenarios:
+Apart from the [`Membrane.Testing.Pipeline`](https://hexdocs.pm/membrane_core/Membrane.Testing.Pipeline.html), which we've already seen, there are a bunch of other testing utilities which may come in handy for different test scenarios:
 
-- [`Membrane.Testing.DataSource`](https://hexdocs.pm/membrane_core/Membrane.Testing.DataSource.html) - Can be used as an alternative for `File.Source` allowing you to pass in an list of payloads which will be supplied to the pipeline.
-- [`Membrane.Testing.Source`](https://hexdocs.pm/membrane_core/Membrane.Testing.Source.html) - Will output data based on the `actions_generator`. Can be useful for generating sequential payloads or a random input.
-- [`Membrane.Testing.Sink`](https://hexdocs.pm/membrane_core/Membrane.Testing.Sink.html) - A fake sink element that will pass all received buffers to a given process. Useful for asserting output buffers one by one.
+- [`Membrane.Testing.Assertions`](https://hexdocs.pm/membrane_core/Membrane.Testing.Assertions.html) - Contains all available assertions that work alongside `Membrane.Testing.Pipeline` and `Membrane.Testing.Sink`.
```

Comment:
`alongside` is not a proper word here (it implies that assertions work separately from these elements, not together with)

Reply:
Bump

Reply:
Fixed, replace alongside with with.